src/Analyser/NodeScopeResolver.php

```diff
@@ -1465,6 +1465,24 @@ function (MutatingScope $scope) use ($expr, $nodeCallback, $context): Expression
 			) {
 				$scope = $scope->assignVariable('http_response_header', new ArrayType(new IntegerType(), new StringType()));
 			}
+
+			if (
+				isset($functionReflection)
+				&& $functionReflection->getName() === 'extract'
+				&& $scope->getType($expr->args[0]->value) instanceof ConstantArrayType
+				&& in_array(count($expr->args), [1, 2], true) // currently only understands simple extract usage
+			) {
+				$flags = $scope->getType($expr->args[1]->value);
+
+				if ($flags instanceof ConstantIntegerType && $flags->getValue() === 0) {
```

Comment:
It might be a better idea to use the actual constant instead of a hardcoded integer. The constant value might change between PHP versions.